File: javaScript/cajero.js
class Cajero {
    constructor(nombre, numeroCuenta, saldo) {
        this.nombre = nombre;
        this.numeroCuenta = numeroCuenta;
        this.saldo = saldo;
    }

    // Función para realizar depósitos
    Deposito(cantidad) {
        this.saldo += cantidad;
        console.log(`Has guardado: ${cantidad}. Nuevo saldo: ${this.saldo}`);
    }

    // Función para realizar retiros
    Retiros(cantidad) {
        if (cantidad > this.saldo) {
            console.log("Fondos insuficientes...");
        } else {
            this.saldo -= cantidad;
            console.log(`Has retirado: ${cantidad}. Nuevo saldo: ${this.saldo}`);
        }
    }

    // Función para realizar transferencias
    Transferencias(cantidad, cuentaDestino) {
        if (cantidad > this.saldo) {
            console.log("Fondos insuficientes...");
        } else {
            this.saldo -= cantidad;
            cuentaDestino.Deposito(cantidad);
            console.log(`Has transferido a ${cuentaDestino.nombre} la cantidad: ${cantidad}. Saldo actual: ${this.saldo}`);
        }
    }
}

// Función principal
function main() {
    // Solicitar datos
    const nombre = prompt("Ingresa tu nombre:");
    const numeroCuenta = prompt("Ingresa tu número de cuenta:");
    const saldo = parseFloat(prompt("Ingresa tu saldo inicial:"));

    const cajero = new Cajero(nombre, numeroCuenta, saldo);

    while (true) {
        console.log("\nOpciones:");
        console.log("1. Depósitos.");
        console.log("2. Retiros.");
        console.log("3. Transferencias.");
        console.log("4. Ver saldo.");
        console.log("5. Salir.");

        const opcion = prompt("Selecciona una opción:");

        if (opcion === '1') {
            const cantidad = parseFloat(prompt("Ingresa la cantidad a ingresar:"));
            cajero.Deposito(cantidad);
        } else if (opcion === '2') {
            const cantidad = parseFloat(prompt("Ingresa la cantidad a retirar:"));
            cajero.Retiros(cantidad);
        } else if (opcion === '3') {
            const cantidad = parseFloat(prompt("Ingresa la cantidad a transferir:"));
            const numeroCuentaDestino = prompt("Ingresa el número de la cuenta de destino:");
            const cuentaDestino = new Cajero("Usuario Destino", numeroCuentaDestino, 0);
            cajero.Transferencias(cantidad, cuentaDestino);
        } else if (opcion === '4') {
            console.log(`Tu saldo es: ${cajero.saldo}`);
        } else if (opcion === '5') {
            console.log("Gracias por usar el cajero automático.");
            break; // Salir del bucle
        } else {
            console.log("Opción no válida, intenta de nuevo.");
        }
    }
}

// Llamar a la función principal
main();

File: python/cajero.py
#en este ejercicio realizaremos un sistema de cajeros sisples en donde tengamos la funciones mas basicas de un cajero

#Creamos la clase de cajero

class Cajero: 
      #definimos la funcion de inicio para la toma de datos iniciales
    def __inicio__ (self, nombre, numeroCuenta, saldo):
        self.nombre = nombre
        self.numeroCuenta = numeroCuenta
        self.saldo = saldo
      # definimos la funcion para los depositos o ingresos de dinero  
    def Deposito(self, cantidad):
        self.saldo += cantidad
        print(f"Has Guardado. " , cantidad , " Nuevo Saldo: ", self.saldo)
     # definimos la funcion para los retiros   
    def Retiros(self, cantidad):
        if cantidad > self.saldo:
            print("Fondos insuficientes...")
        else:
            self.saldo -= cantidad
            print(f"Has Retirado. " , cantidad , " Nuevo Saldo: ", self.saldo)
      #definimos la funcion para realizar las trnsferecnias      
    def Transferencias(self,cantidad, cuenta_destino):
        if cantidad >self.saldo:
            print("Fondos insuficientes...")
        else:
            self.saldo -= cantidad
            cuenta_destino.Deposito(cantidad)
            print(f"Has Transferido a . " , cuenta_destino , " El siguiente dinero: ", cantidad , "Saldo Actual. ", self.saldo)
            
#Aqui realizaremos la funcion principal

def main():
    #Solicitmos datos
    nombre = input("Ingresa tu nombre:  ")
    numeroCuenta  = input("Ingresa tu numero de cuenta:  ")
    saldo =float(input("Ingresa tu saldo Inicial:  "))
      
    Cajero = Cajero(nombre, numeroCuenta, saldo)
    
    while True:
        print("\n Opciones: ")
        print("1. Depositos.")
        print("2. Retiros.")
        print("3. Transferencias.")
        print("4. Ver Saldo.")
        print("5. Salir.")
        
        opcion = input("Selecciona una opcion: ")
        
        if opcion == '1':
            cantidad = float(input("Ingresa la cantidad a ingresar: "))
            Cajero.Depositos(cantidad)
        elif opcion =='2':
            cantidad = float(input("Ingresa la cantidad a retirar: "))
            Cajero.Retiros(cantidad)
        elif opcion =='3':
            cantidad = float(input("Ingresa la cantidad a transferir: "))
            numeroCuentaDestino = input("Ingresa el numero de la cuenta de destino: ")
            cuentaDestino = Cajero("Usuario Destino", numeroCuentaDestino,0)
            Cajero.TranferenciasDinero(cantidad, cuentaDestino)
            
        elif opcion=='4':
            print("Tu Saldo es: ....",Cajero.saldo)
        elif opcion=='5':
            print("Gracias por usar el cajero automatico.")
        else: 
            print("Opcion no valida, Intente de Nuevo.")
            
if Cajero.__inicio__ == "__main__":
    main()
    